code/__HELPERS/icons.dm

```diff
@@ -872,6 +872,52 @@ The _flatIcons list is a cache for generated icon files.
 	del(atom_icon)
 	return text_image
 
+// Pick a random animal instead of the icon, and use that instead
+/proc/getRandomAnimalImage(atom/A)
+	var/list/icons = list()
+	var/icon/selected
+
+	icons += new/icon('icons/mob/animal.dmi', "cow")

```

Comment:
You must not be aware of how shit the ```/icon``` type is, it carries with it quite a high performance cost because it contains all the information of the image state (which is why it's able to be blended/coloured etc), you're making 30 of these, EVERY TIME SOMETHING COMES INTO EXISTENCE.

It may very well be more work, but it makes everything about it better.